docs/bluetooth_fetching_data.md

```diff
@@ -0,0 +1,396 @@
+
+- [`DataUpdateCoordinator`](integration_fetching_data)
+
+## BluetoothProcessorCoordinator
+
+The `ActiveBluetoothProcessorCoordinator` and `PassiveBluetoothProcessorCoordinator` significantly reduce the code needed for creating integrations that primary function as sensor, binary sensors, or fire events. By formatting the data fed into the processor coordinators into a `PassiveBluetoothDataUpdate` object, the
+frameworks can care of creating the entities on demand and allow for minimal `sensor` and `binary_sensor` platform implementations.
+
+These frameworks require the data coming from the library to be formatted into a `PassiveBluetoothDataUpdate` as shown below:
+
+```python
+@dataclasses.dataclass(frozen=True)
+class PassiveBluetoothEntityKey:
+    """Key for a passive bluetooth entity.
+
+    Example:
+    key: temperature
+    device_id: outdoor_sensor_1
+    """
+
+    key: str
+    device_id: str | None
+
+@dataclasses.dataclass(frozen=True)
+class PassiveBluetoothDataUpdate(Generic[_T]):
+    """Generic bluetooth data."""
+
+    devices: dict[str | None, DeviceInfo] = dataclasses.field(default_factory=dict)
+    entity_descriptions: Mapping[
+        PassiveBluetoothEntityKey, EntityDescription
+    ] = dataclasses.field(default_factory=dict)
+    entity_names: Mapping[PassiveBluetoothEntityKey, str | None] = dataclasses.field(
+        default_factory=dict
+    )
+    entity_data: Mapping[PassiveBluetoothEntityKey, _T] = dataclasses.field(
+        default_factory=dict
+    )
+```
+
+### PassiveBluetoothProcessorCoordinator
+
+Example `async_setup_entry` for an integration `__init__.py` using a `PassiveBluetoothProcessorCoordinator`:
+
+```python
+import logging
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.components.bluetooth import BluetoothScanningMode
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothProcessorCoordinator,
+)
+from .const import DOMAIN
+from homeassistant.const import Platform
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+from your_library import DataParser
+
+_LOGGER = logging.getLogger(__name__)
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up example BLE device from a config entry."""
+    address = entry.unique_id
+    data = DataParser()
+    coordinator = hass.data.setdefault(DOMAIN, {})[
+        entry.entry_id
+    ] = PassiveBluetoothProcessorCoordinator(
+        hass,
+        _LOGGER,
+        address=address,
+        mode=BluetoothScanningMode.ACTIVE,
+        update_method=data.update,
+    )
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    entry.async_on_unload(
+        coordinator.async_start()
+    )  # only start after all platforms have had a chance to subscribe
```

Comment:
Has this comment been formatted away from the line above?

Reply:
adjusted